### PR TITLE
BAEL-5131: constructor specification is added c

### PR DESCRIPTION
**Pull request by arash691**

### CODE REVIEW

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Employee.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.baeldung.constructorspecification.rules;
+
+/**
+ * Created by arash on 16.12.21.
+ */
+
+class Employee extends Person {
+    int id;
```
Comment (eric-martin) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Employee.java:8]: blank line after the var decl

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Employee.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    public Employee() {
+        super();
+    }
```
Comment (eric-martin) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Employee.java:11]: blank line between methods

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Person.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+class Person {
+    String name;
+    public Person() {
+        this("Arash");
+    }
+    public Person(String name) {
+        this.name = name;
+    }
+}
+
```
Comment (eric-martin) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Person.java:16]: remove trailing blank lines

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Person.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package com.baeldung.constructorspecification.rules;
+
+/**
+ * Created by arash on 16.12.21.
+ */
+
+class Person {
```
Comment (eric-martin) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Person.java:7]: blank line between variables and methods
blank line between methods

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Test.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.baeldung.constructorspecification.rules;
+
+/**
+ * Created by arash on 16.12.21.
+ */
+
+class Test {
+    public Test(){
```
Comment (eric-martin) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Test.java:8]: Space between () and {

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/simple/Person.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package com.baeldung.constructorspecification.simple;
+
+/**
+ * Created by arash on 16.12.21.
+ */
+
+class Person{
```
Comment (eric-martin) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/simple/Person.java:7]: space between () and {

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/superclass/Employee.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.baeldung.constructorspecification.superclass;
+
+/**
+ * Created by arash on 16.12.21.
+ */
+
+class Employee extends Person{
```
Comment (eric-martin) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/superclass/Employee.java:7]: similar problems here

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/superclass/Person.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package com.baeldung.constructorspecification.superclass;
+
+/**
+ * Created by arash on 16.12.21.
+ */
+
+class Person{
```
Comment (eric-martin) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/superclass/Person.java:7]: Same here

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Employee.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+    public Employee(String name) {
+        super(name);
+    }
+
+    public Employee(int id) {
+        this();
+        //super("John"); // syntax error
```
Comment (andrzejsydor) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/Employee.java:21]: Could be removed?

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/RecursiveConstructorInvocation.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
+package com.baeldung.constructorspecification.rules;
+
+/**
+ * Created by arash on 16.12.21.
+ */
+
```
Comment (andrzejsydor) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/rules/RecursiveConstructorInvocation.java:6]: remove line

File: core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/superclass/Person.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package com.baeldung.constructorspecification.superclass;
+
+/**
+ * Created by arash on 16.12.21.
```
Comment (andrzejsydor) [inline core-java-modules/core-java-lang-oop-constructors/src/main/java/com/baeldung/constructorspecification/superclass/Person.java:4]: It's not needed,the information is in the git

**Pull request merged**